CSharp/DevVmPowershell/Helpers/Implementations/InstanceSettingsHelper.cs

```diff
@@ -125,5 +129,37 @@ public string GetInstanceSettingValue(string name, string section)
 				throw new Exception("Failed to Get Instance Setting Value", ex);
 			}
 		}
+
+		public int GetInstanceSettingArtifactIdByName(string name, string section)
+		{
+			try
+			{
+				int artifactId = 0;
+				using (IInstanceSettingManager instanceSettingManager = ServiceFactory.CreateProxy<IInstanceSettingManager>())
+				{
+					Query query = new Query();
+					query.Condition = $"'Section' == '{section}' AND 'Name' == '{name}'";
+					InstanceSettingQueryResultSet instanceSettingQueryResultSet = instanceSettingManager.QueryAsync(query).Result;
+					if (instanceSettingQueryResultSet.Success)
+					{
+						if (instanceSettingQueryResultSet.TotalCount == 0)
+						{
+							return 0;
```

Comment:
Depending on how this is used, should we leave it at returning 0, or should we throw an error?